readme.md

```diff
@@ -39,6 +39,14 @@ details.branchName // is null if the repository in detached HEAD mode
 details.isCleanTag
 ```
 
+You can optionally search a subset of tags with `prefix`. Inclusion of the `prefix` in the returned version is toggled
+with `includePrefix` (default: `false`). Example when the tag is my-product2.15.0:
+
+```groovy
+gitVersion(prefix:'my-product') // -> 2.15.0
+gitVersion(prefix:'my-product', includePrefix:true) // -> my-product2.15.0
```

Comment:
i'd prefer to be a little more opinionated/constraining here. in particular, i don't want projects to tag `v3.1.0` in order to produce version `3.1.0` with `gitVersion(prefix:'v')`. further, do we need to support both keeping-the-prefix and stripping-the-prefix modes?

proposal: force prefixes to match exactly `[a-z]+-`.

Reply:
What's the reasoning behind the prefix regex of `[/@]?([A-Za-z]+[/@-])+` please?
If we mandate a delimiter after the prefix (i.e. `[/@-]`), then why should we not allow a numeric value?

Use case: I have 2 modules in a Gradle build (`mod`, `mod2`) and I want a prefix to distinguish between the 2 modules.  I can't change the module names, that's not up to me; btw, `mod2` is a newer/better version of `mod`.

Why could the regex not be `[/@]?([A-Za-z0-9]+[/@-])+` ?

Reply:
I think we can allow numbers.